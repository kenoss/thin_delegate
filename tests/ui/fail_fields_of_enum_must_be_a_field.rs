// TODO: Make `register()` is usable for trait definition.
mod private_for_thin_delegate {
    #[thin_delegate::register(AsRef)]
    pub trait AsRef<T: ?Sized> {
        /// Converts this type into a shared reference of the (usually inferred) input type.
        fn as_ref(&self) -> &T;
    }
}

#[thin_delegate::derive_delegate(AsRef<str>)]
enum Named {
    Named {
        x: String,
        y: String,
    },
}

#[thin_delegate::derive_delegate(AsRef<str>)]
enum Unnamed {
    Unnamed(String, String),
}

#[thin_delegate::derive_delegate(AsRef<str>)]
enum Unit {
    Unit,
}

fn main() {}
