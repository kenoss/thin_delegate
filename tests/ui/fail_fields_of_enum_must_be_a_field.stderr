error: fields of enum variant must be a field
  --> tests/ui/fail_fields_of_enum_must_be_a_field.rs:12:11
   |
12 |       Named {
   |  ___________^
13 | |         x: String,
14 | |         y: String,
15 | |     },
   | |_____^

error: fields of enum variant must be a field
  --> tests/ui/fail_fields_of_enum_must_be_a_field.rs:20:12
   |
20 |     Unnamed(String, String),
   |            ^^^^^^^^^^^^^^^^

error: fields of enum variant must be a field
  --> tests/ui/fail_fields_of_enum_must_be_a_field.rs:25:5
   |
25 |     Unit,
   |     ^^^^
